### PR TITLE
make REPL printing of `nothing` an `empty string`

### PR DESCRIPTION
**Pull request by pdeffebach**
Also changes the name of `nastring` to `missingstring`

### CODE REVIEW

**Comment by pdeffebach** [comment]
Actually, I think now that the better way to go about this is to add a method for `ourshowcompact`, but let me know if that's the way forward.

**Comment by pdeffebach** [comment]
Actually, why didn't we just overload `ourshowcompact` to have the `prettyprint` call for `missing` too?

**Comment by pdeffebach** [comment]
Okay now the overall diff should just be one line: an added `ourshowcompact` method.

**Comment by bkamins** [comment]
Maybe one thing - as test coverage has dropped - is this change covered by test cases?

**Comment by pdeffebach** [comment]
Just added some tests.

File: src/abstractdataframe/show.jl [hunk 1]
```diff
@@ -57,6 +57,7 @@ end
 ourshowcompact(io::IO, x::Any) = show(IOContext(io, :compact=>true), x) # -> Void
 ourshowcompact(io::IO, x::AbstractString) = escape_string(io, x, "") # -> Void
 ourshowcompact(io::IO, x::Symbol) = ourshowcompact(io, string(x)) # -> Void
+ourshowcompact(io::IO, x::Nothing) = ourshowcompact(io, "") # -> Void
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]: Couldn't we just do nothing and return `nothing`? Calling `ourshowcompact` on an empty strings sounds overkill for a no-op.

File: test/show.jl [hunk 2]
```diff
@@ -205,6 +205,21 @@ module TestShow
...
+    str = String(take!(io))
+    @test str == """
+    3×2 DataFrames.DataFrame
+    │ Row │ A   │ B         │
+    ├─────┼─────┼───────────┤
+    │ 1   │ 1.0 │ something │
+    │ 2   │ 2.0 │ g         │
+    │ 3   │ 3.0 │           │"""
+
+
```
Comment (nalimilan) [inline test/show.jl]: One empty line is enough.

File: src/abstractdataframe/show.jl (hunk 1 above)
Comment (pdeffebach) [reply src/abstractdataframe/show.jl]: That's a good idea. We don't need to print anything, just define `strlen` to be zero.

File: src/abstractdataframe/show.jl [hunk 3]
```diff
@@ -249,7 +249,10 @@ function showrowindices(io::IO,
                 s = df[i, j]
                 strlen = ourstrwidth(s)
                 if ismissing(s)
-                    printstyled(io, s, color=:light_black)
+                    printstyled(io, "m", color=:light_black)
```
Comment (nalimilan) [inline src/abstractdataframe/show.jl]: This changes the printing of `missing`, right? I'd rather keep it as-is, since `m` isn't a real Julia object.

Comment (pdeffebach) [reply src/abstractdataframe/show.jl]: sorry, that was just me playing around. can you reject that change or should I submit another PR?

**Comment by nalimilan** [comment]
Again, we forgot that the docs should be updated... :-)

**Pull request merged**